backend/src/common/paginate.ts

```diff
@@ -0,0 +1,30 @@
+import { PageDto, PageMetaDto, PageOptionsDto } from '@dtos/pages';
+import { FindOptionsOrder, FindOptionsWhere } from 'typeorm';
+
+interface Identifiable {
+  id: number;
+}
+
+interface FindAndCountable {
+  findAndCount: any;
+}
+
+export async function paginate<
+  U extends FindAndCountable,
+  T extends Identifiable,
+>(
+  repository: U,
+  query: FindOptionsWhere<T>,
+  orderOptions: FindOptionsOrder<T>,
+  pageOptions: PageOptionsDto,
+): Promise<PageDto<T>> {
```

Comment:
```suggestion
export async function paginate<
  Repository extends FindAndCountable<Entity>,
  Entity extends Identifiable,
>(
  repository: Repository,
  query: FindOptionsWhere<Entity>,
  orderOptions: FindOptionsOrder<Entity>,
  pageOptions: PageOptionsDto,
): Promise<PageDto<Entity>> {
```

Reply:
Thank you, these suggestions improve the readability a lot.

I realized I could eliminate the second template parameter. See what you think, but I hope it makes it cleaner than what I had before.

Reply:
I think you are right, it is cleaner.